src/main/java/com/github/jnidzwetzki/bitfinex/v2/command/OrderMultiCommand.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ *
+ *    Copyright (C) 2015-2018 Jan Kristof Nidzwetzki
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ *******************************************************************************/
+package com.github.jnidzwetzki.bitfinex.v2.command;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Preconditions;
+
+import com.github.jnidzwetzki.bitfinex.v2.BitfinexWebsocketClient;
+import com.github.jnidzwetzki.bitfinex.v2.exception.BitfinexCommandException;
+
+public class OrderMultiCommand implements BitfinexOrderCommand {
+
+    private final static Pattern REGEX = Pattern.compile("0,\\s*(.*),\\s*null", Pattern.DOTALL);
```

Comment:
I'm not a big fan of the direct manipulation of JSON by using a regex. When somebody changes the JSON (e.g., reformat it) the regex might fail. What do you think about one of the following solutions:

* Parsing the result of the sub-commands by using the `JSONObject` and extracting the needed data directly by operating on the JSON data. 

* Changing the method `public String getCommand` of the `BitfinexCommand` interface to `public JSONObject getCommand`. Then we could extract the needed JSON part directly from the result without using a regex or parsing it again (this is a larger refactoring).

Reply:
Hi Jan, I am not fan of REGEX manipulations as well.
* Parsing the result of the sub-commands by using the JSONObject and extracting the needed data directly by operating on the JSON data.

Feels a little bit like a hack - and it's not solving the problem (keep reading)
* Changing the method public String getCommand of the BitfinexCommand interface to public JSONObject getCommand. Then we could extract the needed JSON part directly from the result without using a regex or parsing it again (this is a larger refactoring).

This one I was having in mind right from the start, however - in the end, none of proposed solutions is solving the problem that we have right now - In case of someone changing the "inner" JSON (which is us only) - this step will fail too. JSON is serializing "array" not "map", so we would need to fetch data by indices, not keys.
REGEX is plain simple in its behaviour, so there is no way it will fail at any point.

The only thing we can do about it, is keeping proper tests around this step.
It's up to you :) there is no good option that we have here.